src/main/java/net/rptools/maptool/client/functions/LookupTableFunction.java

```diff
@@ -84,367 +90,521 @@ public Object childEvaluate(
+   * This is used in the PickOnce family of tables.
+   *
+   * <ul>
+   *   <li>{@code resetTablePicks(tblName)} - reset all entries on a table
+   *   <li>{@code resetTablePicks(tblName, entriesToReset)} - reset specific entries from a String
+   *       List with "," delim
+   *   <li>{@code resetTablePicks(tblName, entriesToReset, delim)} - use custom delimiter
+   *   <li>{@code resetTablePicks(tblName, entriesToReset, "json")} - entriesToReset is a JsonArray
+   * </ul>
+   *
+   * @param function name of the MTscript function (used for error messages)
+   * @param params list of function parameters
+   * @throws ParserException Thrown when an invalid or missing parameter is supplied
+   */
+  private @NotNull String resetTablePicks(String function, List<Object> params)
+      throws ParserException {
+    checkTrusted(function);
+    FunctionUtil.checkNumberParam(function, params, 1, 3);
+    String tblName = params.get(0).toString();
+    LookupTable lookupTable = checkTableAccess(tblName, function);
+    if (params.size() > 1) {
+      String delim = (params.size() > 2) ? params.get(2).toString() : ",";
+      List<String> entriesToReset;
+      if (delim.equalsIgnoreCase("json")) {
+        JsonArray jsonArray = FunctionUtil.paramAsJsonArray(function, params, 1);
+        entriesToReset =
+            JSONMacroFunctions.getInstance()
+                .getJsonArrayFunctions()
+                .jsonArrayToListOfStrings(jsonArray);
       } else {
-        lookupTable.reset();
+        entriesToReset = StrListFunctions.toList(params.get(1).toString(), delim);
       }
-      MapTool.serverCommand().updateCampaign(MapTool.getCampaign().getCampaignProperties());
-      return "";
-    } else if ("setTablePickOnce".equalsIgnoreCase(function)) {
+      lookupTable.reset(entriesToReset);
+    } else {
+      lookupTable.reset();
+    }
+    MapTool.serverCommand().updateCampaign(MapTool.getCampaign().getCampaignProperties());
+    return "";
+  }
 
-      checkTrusted(function);
-      FunctionUtil.checkNumberParam("setTablePickOnce", params, 2, 2);
-      String name = params.get(0).toString();
-      String pickonce = params.get(1).toString();
-      LookupTable lookupTable = getMaptoolTable(name, function);
-      lookupTable.setPickOnce(FunctionUtil.getBooleanValue(pickonce));
-      MapTool.serverCommand().updateCampaign(MapTool.getCampaign().getCampaignProperties());
-      return lookupTable.getPickOnce() ? BigDecimal.ONE : BigDecimal.ZERO;
+  private @NotNull Object getTableEntry(String function, List<Object> params)
+      throws ParserException {
+    FunctionUtil.checkNumberParam(function, params, 2, 2);
+    String name = params.get(0).toString();
+    LookupTable lookupTable = checkTableAccess(name, function);
+    String roll = params.get(1).toString();
+    LookupEntry entry = lookupTable.getLookupDirect(roll);
+    if (entry == null) return "";
```

Comment:
Does not follow formatting standards. The same formatting standards issue crops up multiple places in this change

Reply:
Hm, shouldn't spotlessApply take care of the formatting?  Running spotless was the last commit in the PR.

(Or maybe there are things spotless doesn't correct/change and must be done manually?  If so, is there a list of such things that spotless doesn't fix but it should?  Apparently single-line IF statements is one of them; what about using braces when a true or false block only contains a single statement?)

Reply:
The Google formatting library that Spotless uses doesn't deal with single statement ifs because of reasons... (to keep Android developers happy as the coding guidelines for Android are different on that one point).

There may be others but they haven't come up before (this one has come up several times)